src/gateway/InMemoryGameGateway.ts

```diff
@@ -0,0 +1,28 @@
+import { Game } from "../domain/game";
+import { GameGateway } from "./GameGateway";
+
+export class InMemoryGameGateway implements GameGateway {
+    private allGames : Game[];
+
+    constructor() {
+        this.allGames = new Array();
+    }
+
+    addGameToAllGamesArray(newGame : Game): void {
+        this.allGames.push(newGame);
+    }
+
+    getGameById(playerIdToLookFor : number): Game {
+        return this.findSpecificGame(playerIdToLookFor);
+    }
+
+    private findSpecificGame(playerId : number) : Game {
+        var test = new Game(-1, "");
```

Comment:
visas šitas kodas gali virsti į `return this.allGames.filter(game => game.getPlayerId() === playerId) ?? new Game(-1, "");`